### PR TITLE
Mr rating select

### PR DESCRIPTION
**Pull request by merosenlund**
Updated the Rating Stars component in the Add Review Form so that it has a hover functionality as well as an onclick functionality.

[Trello Ticket](https://trello.com/c/6FqJyUsY)

Functionality

### CODE REVIEW

**Comment by merosenlund** [comment]
[Add onhover ability to the stars component](https://trello.com/c/6FqJyUsY/177-add-onhover-ability-to-the-stars-component)

File: src/components/reviews/ReviewsList.jsx [hunk 1]
```diff
@@ -8,7 +8,25 @@ import { useProductContext } from '../../context/ProductContext.jsx';
...
-  const reviewsList = reviews.results;
+  const [sortBy, setSortBy] = useState('relevance');
+  const reviewsList = reviews.results.slice();
+
+  reviewsList.sort((a, b) => {
+    // The daysSince helper function is at the bottom of the file
+    let aDaysAgo = daysSince(a.date);
+    let bDaysAgo = daysSince(b.date);
+
+    if (sortBy === 'relevance') {
```
Comment (vinhle000) [inline src/components/reviews/ReviewsList.jsx:19]: Thanks for the comment about comparison weight for this sorting implmentation

File: src/components/reviews/ReviewForm.jsx [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
 import styles from './reviews.module.css';
 
 
 const ReviewForm = ({close}) => {
+  const [hoverRating, setHoverRating] = useState(0)
+  const [rating, setRating] = useState(null)
+
+  const onRatingHover = (e) => {
+    const target = e.target;
+    const rect = target.getBoundingClientRect();
```
Comment (vinhle000) [inline src/components/reviews/ReviewForm.jsx:13]: Is this function provided by React events?

File: src/components/reviews/ReviewForm.jsx [hunk 3]
```diff
@@ -1,17 +1,50 @@
...
+    // This gets where the mouse is pointing in the target element on the x axis
+    const x = e.clientX - rect.left;
```
Comment (vinhle000) [inline src/components/reviews/ReviewForm.jsx:15]: Also is clientX a property of React events?

**Review by vinhle000 — APPROVED** [review]
Looks good! Was just curious about the 'getBoundingClientRect' method and the 'clientX' property of the event.

**Pull request merged**